scipy/special/tests/test_hyp2f1.py

```diff
@@ -0,0 +1,608 @@
+            ),
+            pytest.param(
+                Hyp2f1TestCase(
+                    a=2.02764642551431,
+                    b=16.088264119063613,
+                    c=8.031683612216888,
+                    z=(0.5263157894736841-0.5263157894736843j),
+                    expected=(-0.6906825165778091+0.8176575137504892j),
+                    rtol=5e-13,
+                ),
+            ),
+        ]
+    )
+    def test_region1(self, hyp2f1_test_case):
+        """|z| < 0.9 and real(z) >= 0."""
+        a, b, c, z, expected, rtol = hyp2f1_test_case
+        assert abs(z) < 0.9 and z.real >= 0  # Tests the test
+        assert_allclose(hyp2f1(a, b, c, z), expected, rtol=rtol)
+
+    @pytest.mark.parametrize(
+        "hyp2f1_test_case",
+        [
+            pytest.param(
+                Hyp2f1TestCase(
+                    a=2.02764642551431,
+                    b=1.0561196186065624,
+                    c=4.078873014294075,
+                    z=(-0.3157894736842106+0.7368421052631575j),
+                    expected=(0.7751915029081136+0.24068493258607315j),
+                    rtol=5e-15,
+                ),
+            ),
+        ]
+    )
+    def test_region2(self, hyp2f1_test_case):
+        """|z| < 1 and real(z) < 0."""
+        a, b, c, z, expected, rtol = hyp2f1_test_case
+        assert abs(z) < 1 and z.real < 0  # Tests the test
+        assert_allclose(hyp2f1(a, b, c, z), expected, rtol=rtol)
+
+    @pytest.mark.xfail(reason="gh-8054")
+    @pytest.mark.parametrize(
+        "hyp2f1_test_case",
+        [
+            pytest.param(
+                Hyp2f1TestCase(
+                    a=1.0272592605282642,
+                    b=1.0561196186065624,
+                    c=-0.906685989801748,
+                    z=(0.10526315789473673-0.9473684210526316j),
+                    expected=(-3.9995506969395858-8.179533155338005j),
+                    rtol=1e-12,
+                ),
+            ),
+            pytest.param(
+                Hyp2f1TestCase(
+                    a=8.5,
+                    b=4.5,
+                    c=8.077282662161238,
+                    z=(0.3157894736842106+0.9473684210526314j),
+                    expected=(-0.11307039404123598-0.443195310438102j),
+                    rtol=1e-12,
+                ),
+            ),
+        ]
+    )
+    def test_region4(self, hyp2f1_test_case):
+        """0.9 <= |z| <= 1 and |1 - z| >= 1.
+
+        This region is unhandled by of the standard transformations and
+        needs special care.
+        """
+        a, b, c, z, expected, rtol = hyp2f1_test_case
+        assert 0.9 <= abs(z) <= 1 and abs(1 - z) >= 1  # Tests the test
+        assert_allclose(hyp2f1(a, b, c, z), expected, rtol=rtol)
+
+    # Marked as slow so it won't run on continuous integration builds.
+    # This test is not actually that slow, though could become slow in
+    # the future if enough test cases are added.
+    @pytest.mark.slow
```

Comment:
Note that this will run on CI jobs that have `full` in the name - and the Windows full test suite run is already at around the 60 min timeout limit. There's a `pytest.mark.xslow` you want to use instead.

Reply:
Thanks, I didn't realize that. I'll change it.  The entire test suite currently runs in 0.39 seconds on my machine, but this last test could creep up in running time as more test cases are added. I added `test_test_hyp2f1.py` at the end because I was worried reviewers could get bogged down trying to check the validity of each test case individually. I wanted there to be a convenient way for reviewers to check all of the cases. The problem then reduces to checking that the logic to gather the test cases out of the pytest marks is correct.

Reply:
I've looked into it, and I think this test actually should be marked slow, not xslow. Code authors and reviewers would not want it to be skipped,and it's reasonable to run in full builds. It should be skipped by default. It's status is similar to the tests marked slow in `test_mpmath.py`. The running time should never exceed the 1-10 second range.

Reply:
Okay, thanks for looking into that!